### ISSUE TITLE
some idea for findLastIndex

### ISSUE THREAD

**Issue by fatedeity** [30-seconds/30-seconds-of-code#1005]
## Description

```javascript
const findLastIndex = (arr, fn) =>
  arr
    .map((val, i) => [i, val])
    .filter(([i, val]) => fn(val, i, arr))
    .pop()[0];

// example
findLastIndex([1, 2, 3, 4], n => n % 2 === 1); // 2

findLastIndex([1, 2, 3, 4], n => n === 5);
// Uncaught TypeError: Cannot read property '0' of undefined
//  at findLastIndex (<anonymous>:5:11)
// at <anonymous>:1:1
```
For a better experience, add a default value.
```javascript
const findLastIndex = (arr, fn) =>
  (arr
    .map((val, i) => [i, val])
    .filter(([i, val]) => fn(val, i, arr))
    .pop() || [-1])[0];

// example
findLastIndex([1, 2, 3, 4], n => n === 5); // -1
```

**Comment by fejes713** [comment]
Hey @fatedeity 

To me, this sounds like a nice improvement from the UX standpoint. Would you like to open a PR by yourself and enhance this? 

I'd suggest taking a look at our [contributing docs](https://github.com/30-seconds/30-seconds-of-code/blob/master/CONTRIBUTING.md). Let me know if you need help with anything!

**Comment by Chalarangelo** [comment]
Agreed, this seems like a nice improvement. I'd like to have this merged if you submit a PR for it.

**Comment by Chalarangelo** [comment]
Resolved in #1006
